pandas/tests/frame/test_constructors.py

```diff
@@ -2628,7 +2628,9 @@ def test_datetime_date_tuple_columns_from_dict(self):
         # GH 10863
         v = date.today()
         tup = v, v
-        result = DataFrame({tup: Series(range(3), index=range(3))}, columns=[tup])
+        with tm.assert_produces_warning(FutureWarning, check_stacklevel=False):
+            # GH#36131 comparison of Timestamp vs date
+            result = DataFrame({tup: Series(range(3), index=range(3))}, columns=[tup])
```

Comment:
Is there a way to avoid this warning in this case? (it's not something a user can do something about?) 
And will the behaviour change here after the deprecationg has been enforced?

Reply:
any thoughts here @jbrockmendel

Reply:
just the warning not to use date objects, or removing the Categorical behavior of passing convert_dates=True to maybe_infer_to_datetimelike

Reply:
this should not be warning at the top level

Reply:
In this the user/test is passing date objects, and gets a warning that behavior with date objects is going to change.

I can add a test in which the user-facing behavior is going to change (and is currently wrong)

```
v = date.today()
tup = v, v

ts = pd.Timestamp(v) + pd.Timedelta(hours=1)
tup2 = ts, ts

ser = pd.Series(range(3))

result = pd.DataFrame({tup2: ser}, columns=[tup])

>>> result
   (2020-12-17, 2020-12-17)
0                         0
1                         1
2                         2
```

But after this deprecation is enforced, we're going to end up with

```
>>> result
Empty DataFrame
Columns: [(2020-12-17, 2020-12-17)]
Index: []
```

which btw is analogous to what we'd get today if we swapped tup and tup2 above:

```
result2 = pd.DataFrame({tup: ser}, columns=[tup2])

>>> result2
Empty DataFrame
Columns: [(2020-12-17 01:00:00, 2020-12-17 01:00:00)]
Index: []
```